odoo/fields.py

```diff
@@ -5213,7 +5228,7 @@ def write_new(self, records_commands_list):
             ])
 
 
-class Id(Field):
+class Id(Field[int]):
```

Comment:
```suggestion
class Id(Field[int | NewId]):
```
... or else type the needed methods.